src/graph_notebook/magics/graph_magic.py

```diff
@@ -340,7 +342,11 @@ def gremlin(self, line, cell, local_ns: dict = None):
             children.append(table_output)
 
             try:
-                gn = GremlinNetwork()
+                groupby=None
+                if args.group_by and args.group_by != '':
+                    groupby=args.group_by
+                logger.debug(f'groupby: {groupby}')
+                gn = GremlinNetwork(group_by_property=groupby)
```

Comment:
the first statement is likely not needed, since we have set a default for `--group-by` in the argument parser, it will always be present (and default to `T.label`)

This should just be able to be reduced to:

```
logger.debug(f'group_by: {args.group_by}')
gn = GremlinNetwork(group_by_property=group_by)
```
```